js/adapt-contrib-resources.js

```diff
@@ -1,31 +1,15 @@
-define([
-  'core/js/adapt',
-  './adapt-contrib-resourcesView',
-  './adapt-contrib-resourcesHelpers'
-], function(Adapt, ResourcesView, ResourcesHelpers) {
+import Adapt from 'core/js/adapt';
+import ResourcesView from './ResourcesView';
+import ResourcesHelpers from './ResourcesHelpers';
 
-  function setupResources(resourcesData) {
-
-    var resourcesModel = new Backbone.Model(resourcesData);
-    var resourcesCollection = new Backbone.Collection(resourcesModel.get('_resourcesItems'));
-
-    Adapt.on('resources:showResources', function() {
-      Adapt.drawer.triggerCustomView(new ResourcesView({
-        model: resourcesModel,
-        collection: resourcesCollection
-      }).$el);
-    });
-
-  }
-
-  function initResources() {
-
-    var courseResources = Adapt.course.get('_resources');
+class Resources {
```

Comment:
What we've tended to do with these kind of controller classes is make them into a singleton.

https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/headings.js#L17

The class inherits the ability to be an event listener from `Backbone.Controller` and its `initialize` function is executed when the `Backbone.Controller` is instantiated. At the end of the file the singleton is instantiated and returned.